orangecontrib/text/pubmed.py

```diff
@@ -0,0 +1,480 @@
+        for field_name, field_key
+        in includes_metadata
+    ]
+    class_vars = [
+        DiscreteVariable(
+                'section_name',
+                values=list(set(class_values))
+        )
+    ]
+    domain = Domain(
+            [],
+            class_vars=class_vars,
+            metas=meta_vars
+    )
+
+    Y = np.array([class_vars[0].to_val(cv)
+                  for cv
+                  in class_values])[:, None]
+    Y[np.isnan(Y)] = 0
+
+    return Corpus(
+            None,
+            Y,
+            meta_values,
+            domain
+    )
+
+
+class Pubmed:
+
+    DEFAULT_BATCH_SIZE = 1000
+
+    def __init__(self, email, progress_callback=None, error_callback=None):
+        if not validate_email(email):
+            raise ValueError('{} is not a valid email address.'.format(email))
+
+        self.email = email
+        Entrez.email = email
+
+        self.record_id_list = None  # Ids of the records available.
+        self.search_record_count = 0  # Number of records.
+        self.search_record_web_env = None  # BioPython history param.
+        self.search_record_query_key = None  # BioPython history param.
+
+        self.progress_callback = progress_callback
+        self.error_callback = error_callback
+
+        self.cache_path = None
+        cache_folder = os.path.join(environ.buffer_dir, 'pubmedcache')
+        try:
+            if not os.path.exists(cache_folder):
+                os.makedirs(cache_folder)
+            self.cache_path = os.path.join(cache_folder, 'query_cache')
+        except:
+            warnings.warn('Could not assemble Pubmed query cache path',
+                          RuntimeWarning)
+
+    def _search_for_records(self, terms=[], authors=[],
+                            pub_date_start=None, pub_date_end=None):
+        """Executes a search for PubMed records.
+
+        Results of the search, like QueryKey, WebEnv and Count are stored as
+        parameters of this class's instance.
+
+        Args:
+            terms (list): Keywords to search by.
+            authors (list): Authors to search by.
+            pub_date_start (str): The start of the publish date chronological
+                interval.
+            pub_date_end (str): The end of the publish date chronological
+                interval.
+        """
+        search_terms = []
+
+        argument_checks = [
+            {
+                'argument': authors,
+                'error': '{} is not a valid data type for the field "authors".'
+                    .format(type(authors)),
+                'conjunction': ' AND ',
+                'field': 'Author',
+            },
+            {
+                'argument': terms,
+                'error': '{} is not a valid data type for the field "terms".'
+                    .format(type(terms)),
+                'conjunction': ' OR ',
+                'field': 'All fields',
+            },
+        ]
+
+        for check in argument_checks:
```

Comment:
IMO this dicts are hard to follow. I would prefer tuples so instead of getting values from the dict you can work with variables. Something like this:

```
argument_checks = [
    (authors, 'Author', ' AND ',
     '{} is not a valid data type for the field "authors".'
        .format(type(authors)),),
    (terms, 'All fields', ' OR ',
     '{} is not a valid data type for the field "terms".'
     .format(type(terms)),),
]

for argument, field, conjunction, error in argument_checks:
    if not isinstance(argument, list):
        raise ValueError(error)
    arguments_list = ['{}[{}]'.format(arg, field)
                      for arg in argument]
    arguments = conjunction.join(arguments_list)
    if arguments:
        search_terms.append('({})'.format(arguments))
```

What do you prefer?